eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -326,10 +325,10 @@ public synchronized BackupRegistry get() {
             }
 
             DiscoveryJerseyProvider discoveryJerseyProvider = new DiscoveryJerseyProvider(
-                    CodecWrapper.get(CodecType.from(clientConfig.getJsonCodecName())),
-                    CodecWrapper.get(CodecType.from(clientConfig.getXmlCodecName()))
+                    CodecWrappers.getEncoder(clientConfig.getEncoderName()),
```

Comment:
Agree. The public REST API current does not care about codecs, but takes a X-Eureka-Accept header. DiscoveryClient (for now, during the transition) translates between the specified decoder and the appropriate X-Eureka-Accept (translation in EurekaAccept.java)